builtin/credential/aws/path_role.go

```diff
@@ -348,14 +390,117 @@ func (b *backend) pathRoleCreateUpdate(
 		roleEntry.BoundIamInstanceProfileARN = boundIamInstanceProfileARNRaw.(string)
 	}
 
-	// Ensure that at least one bound is set on the role
-	switch {
-	case roleEntry.BoundAccountID != "":
-	case roleEntry.BoundAmiID != "":
-	case roleEntry.BoundIamInstanceProfileARN != "":
-	case roleEntry.BoundIamRoleARN != "":
-	default:
+	if boundIamPrincipalARNRaw, ok := data.GetOk("bound_iam_principal_arn"); ok {
+		roleEntry.BoundIamPrincipalARN = boundIamPrincipalARNRaw.(string)
+	}
+
+	if inferRoleTypeRaw, ok := data.GetOk("role_inferred_type"); ok {
+		roleEntry.RoleInferredType = inferRoleTypeRaw.(string)
+	}
 
+	if inferredAWSRegionRaw, ok := data.GetOk("inferred_aws_region"); ok {
+		roleEntry.InferredAWSRegion = inferredAWSRegionRaw.(string)
+	}
+
+	allowEc2Auth, allowIamAuth := false, false
+
+	parseAllowedAuthTypes := func(input string) string {
+		allowedAuthTypes := []string{}
+		for _, t := range strings.Split(input, ",") {
+			if t == "ec2" {
+				allowedAuthTypes = append(allowedAuthTypes, t)
+			} else if t == "iam" {
+				allowedAuthTypes = append(allowedAuthTypes, t)
+			} else if t != "" {
+				return fmt.Sprintf("unrecognized auth type: '%s'", t)
+			}
+		}
+		roleEntry.AllowedAuthTypes = allowedAuthTypes
+		return ""
+	}
+
+	if allowedAuthTypesRaw, ok := data.GetOk("allowed_auth_types"); ok {
+		err := parseAllowedAuthTypes(allowedAuthTypesRaw.(string))
+		if err != "" {
+			return logical.ErrorResponse(err), nil
+		}
+	} else if req.Operation == logical.CreateOperation {
+		err := parseAllowedAuthTypes(data.Get("allowed_auth_types").(string))
+		if err != "" {
+			return logical.ErrorResponse(err), nil
+		}
+	}
+
+	for _, t := range roleEntry.AllowedAuthTypes {
+		if t == "ec2" {
```

Comment:
OK, that sounds fair. I just wanted to make sure it wasn't just some forgotten copypasta.